File: fixtures/scale/lua/large/checks/check_pulsestream.lua
local pulsestream = require("ciphers.pulsestream")
local pulsestream_spec = require("specs.pulsestream_spec")
local corpus_stream = require("corpus.corpus_stream")

local function check_pulsestream()
  local spec = pulsestream_spec.get()
  for _, text in ipairs(corpus_stream.corpus_stream()) do
    local sealed = pulsestream.pulsestream_encrypt(text, spec.key)
    local opened = pulsestream.pulsestream_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "stream"
end

return { check_pulsestream = check_pulsestream }

File: fixtures/scale/javascript/large/src/checks/checkPulsestream.js
import pulsestream from '../ciphers/pulsestream.js';
import pulsestreamSpecMod from '../specs/pulsestreamSpec.js';
import corpusStreamMod from '../corpus/corpusStream.js';

function checkPulsestream() {
  const spec = pulsestreamSpecMod.pulsestreamSpec();
  for (const text of corpusStreamMod.corpusStream()) {
    const sealed = pulsestream.pulsestreamEncrypt(text, spec.key);
    const opened = pulsestream.pulsestreamDecrypt(sealed, spec.key);
    if (opened !== text) return false;
  }
  return spec.category === 'stream';
}

export default { checkPulsestream };
